docs/draft/rest_api.md

```diff
@@ -1,201 +1,293 @@
 
-### Submitting user-input questions
+*returns:* 201 Created (immediately - note this does not mean
+that the transformation is finished)
+*ERRORS*:  
+423 - Locked: transformation not ready but in state "user-input"
 
-### Submitting a Cloud Service Archive for transformation
+##### DELETE /toscamodels/{id}/transformations/{platform}
+Halts the specified transformation.
 
-A single call containing a CSAR and the target platform
-(for some platforms credentials might be necessary, these will probably be supplied by user input)
-```
-POST transformation/{platform}
-```
+*Postcondition:* Status of specified transformation is "canceled"
 
-**parameters**:
--  **platform**: the platform on which the application should be transformed. (Supported platforms can be retrieved using a status request (capabilities))
+*ERRORS:*  
+404 - transformation doesn't exit (TOSCA model oder platform does not exist)
 
-**sends:**
-blob of the CSAR,
+### Reading transformation logs
+##### GET /toscamodels/{id}/transformations/{platform}/logs/
+Receive the logs for specified transformation. All logs starting with the {start}nth to the most recent log are transfered.
 
-**on success returns:**
-  
-HTTP Response Code: 200
-  
+*Request body:*
 ```json
 {
-  "id": 1
+    "start":0
 }
 ```
-**on error returns:**
-  
-HTTP Error Code: 500
-  
+- start - index of first log to receive
+
+*returns:*
 ```json
 {
-  "message": "Something went wrong.",
-  "error_code": 121341,
-  "log": "log"
+    "end":53,
+    "logs":["line1","line2",...]
 }
 ```
-- *message:* short message including what did not work
-- *error_code*: code to easily identify problem
-- *log*: log for that action
+- end: the index of the last log line
+- logs: array of log lines (order: oldest first)
 
-### Aborting a transformation
-```
-DELETE transformation/{id}
+*ERRORS:*  
+400 - start index out of bounds
+404 - no logs available
+
+*EXAMPLE*:
+1. Client calls GET .../logs?start=0
+2. Server answers with
+```json
+{
+    "end":3,
+    "logs:["line1","line2","line3","line4"]
+}
 ```
-**on success returns :** nothing if abortion was successful (HTTP Response Code 200)
+3. Client calls GET .../logs?start=4
+4. etc
+### Downloading platform artifacts
+##### GET /toscamodels/{id}/transformations/{platform}/artifact
+Downloads the deployment artifact for specified platform and TOSCA model.
+
+*ERRORS:*  
+404 - The artifact does not exist
 
-**on error returns:**
+### Specifying additional user-input
 
-HTTP Response Code: 500
+If the transformation status changes to `user-input` the transformator needs additonal data from the client in order to perform the transformation.
 
+To get information about required data, call:
+```
+GET /toscamodel/{id}/transformations/{platform}/user-input
+```
+*returns:*  
 ```json
-{
-  "message": "Aborting transformation did not work.",
-  "error_code": 121341,
-  "log": "log"
-}
+[
+    "Database Password": {
```

Comment:
You mean without the space character? I thought about allowing spaces in the key values, in order to display them nicely in the GUI! So in the gui client, no string transformation is needed.